File: CSharp/Yow.CoD.Finance.Domain/Contracts/LoanSettledEvent.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class LoanSettledEvent : Event
    {
        public LoanSettledEvent(DateTimeOffset transactionDate)
        {
            TransactionDate = transactionDate;
        }

        public DateTimeOffset TransactionDate { get; }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/contracts/LoanSettledEvent.java
package com.leecampbell.cod.domain.contracts;

import java.time.OffsetDateTime;

public final class LoanSettledEvent extends DomainEvent{
    private final OffsetDateTime transactionDateTime;

    public LoanSettledEvent(OffsetDateTime transactionDateTime) {
        this.transactionDateTime = transactionDateTime;
    }

    public OffsetDateTime transactionDateTime(){
        return this.transactionDateTime;
    }
}